e2e/nx-plugin/src/nx-plugin.test.ts

```diff
@@ -172,6 +177,59 @@ describe('Nx Plugin', () => {
     });
   }, 90000);
 
+  it('should be able to use a local plugin', async () => {
+    const plugin = uniq('plugin');
+    const generator = uniq('generator');
+    const generatedProject = uniq('project');
+
+    runCLI(`generate @nrwl/nx-plugin:plugin ${plugin} --linter=eslint`);
+    runCLI(
+      `generate @nrwl/nx-plugin:generator ${generator} --project=${plugin}`
+    );
+
+    updateFile(
+      `libs/${plugin}/src/index.ts`,
+      `import {basename} from 'path'
+
+export function registerProjectTargets(f) {
```

Comment:
This is interesting! I never knew this was an option until I saw this PR and checked the docs.

Reply:
When we added the project inference capabilities for `package.json`, I followed it up with extending the inference to support nx-plugins. This is pretty cool for non-js plugins to be able to register targets based on say a `csproj` file.

Reply:
Yeah, plugin capabilities in the docs, in general, should be reworked IMHO. It's repeated a few times in different places, but each place misses some of the pictures.  I'll be looking at it sometime soon.